### ISSUE TITLE
Unable to use `content-type` header

### ISSUE THREAD

**Issue by alex-way** [ts-rest/ts-rest#453]
It seems like the `content-type` header is being stripped from my request when passing through:

```typescript
const contract = c.router({
	getAccessToken: {
		method: "POST",
		path: "/oauth_token.do",
		body: z.string(),
		responses: {
			200: z.object({
				access_token: z.string(),
				token_type: z.string(),
				expires_in: z.number(),
				scope: z.string(),
				refresh_token: z.string(),
			}),
		},
		headers: z.object({
			"content-type": z.literal("application/x-www-form-urlencoded"),
		}),
	},
})
```

I understand there's a contentType attribute when creating the route. However, it only has strict options:

![image](https://github.com/ts-rest/ts-rest/assets/29122169/e472c145-85f8-4b96-8514-f50129eee04a)

**Comment by alex-way** [comment]
Additionally it would be cool if we could transform the body to the correct format, using URLEncode from a zod object. But I'll get a separate feature request raised for that.

**Comment by alex-way** [comment]
```typescript
import { initClient, initContract } from "@ts-rest/core";
import z from "zod";

const echoContract = c.router({
	sendPost: {
		method: "POST",
		path: "/post",
		body: z.string(),
		responses: {
			200: z.string(),
		},
		contentType: "application/x-www-form-urlencoded",
	},
});

export const echoClient = initClient(echoContract, {
	baseUrl: "https://postman-echo.com",
	baseHeaders: {
		"content-type": "application/x-www-form-urlencoded",
	},
});


echoClient.sendPost({
	body: "foo=bar&baz=qux",
	extraHeaders: {
		"content-type": "application/x-www-form-urlencoded",
	},
}).then((response) => {
	console.log(response.headers.get("content-type"));
});
```

**Comment by alex-way** [comment]
Further debugging has lead me to believe this is the issue:
https://github.com/ts-rest/ts-rest/blob/9fcd88eb35a91e8a0940e5d6cd06539d710709a1/libs/ts-rest/core/src/lib/client.ts#L227

The reason being is that any request is getting it's body "JSON.parse"'d which means that even if you specify z.string() as the body type. It will convert the string `Hello` to `"Hello"`.

**Comment by michaelangeloio** [comment]
#461 closes this

**Comment by joaolisboa** [comment]
In the new version the body is encoded with FormData but that won't always be the case, or did I miss something with this change?


Example:

```
export const contract = c.router({
  login: {
    method: 'POST',
    path: '/authentication',
    contentType: 'application/x-www-form-urlencoded',
    responses: {
      200: c.type<Response<User>>(),
    },
    body: z.any(),
  },
})
```

```
  const params = new URLSearchParams()
  params.append('username', values.username)
  params.append('password', values.password)

  console.log(params.toString())
  // prints: username=test%40mail.com&password=test

  const data = await client.auth.login({
    body: params.toString(),
  })
```

Result:
![CleanShot 2024-01-20 at 16 36 22@2x](https://github.com/ts-rest/ts-rest/assets/17931192/235820b0-c44b-47ba-b327-d0bf55202413)


What I expected:
![CleanShot 2024-01-20 at 16 37 29@2x](https://github.com/ts-rest/ts-rest/assets/17931192/29ba93f7-4f7a-49f7-b49d-0a3e73b04504)

**Comment by alex-way** [comment]
I think there's a bug somewhere - For some reason ts-rest is JSON encoding the body, which means that a string is turned into: `"my_payload_here"` rather than being `my_payload_here`. I'll get a bug raised an hopefully resolve at some point in the next couple of weeks

**Comment by Gabrola** [comment]
@joaolisboa @alex-way fixed in #524
